contracts/RenPool.sol

```diff
@@ -70,16 +77,47 @@ contract RenPool {
         uint senderBalance = balances[sender];
 
         require(senderBalance > 0 && senderBalance >= _amount, "Insufficient funds");
-        require(isLocked == false, "Pool is locked");
-
-        renToken.transfer(sender, _amount);
+        require(isLocked == false);
         totalPooled -= _amount;
         balances[sender] -= _amount;
-
+        renToken.transfer(sender, _amount);
+            
         emit RenWithdrawal(sender, _amount);
+
+    }
+
+    function withdrawRequest(uint _amount) external {
+        address sender = msg.sender;
```

Comment:
I would call this `requestWithdraw`, we are usually naming our function using verbs `deposit`, `withdraw`, `fullfill...`